bls12_381/src/curves/g1.rs

```diff
@@ -40,6 +43,32 @@ impl SWModelParameters for Parameters {
     fn mul_by_a(_: &Self::BaseField) -> Self::BaseField {
         Self::BaseField::zero()
     }
+
+    fn is_in_correct_subgroup_assuming_on_curve(p: &GroupAffine<Parameters>) -> bool {
+        // Check that sigma(P) == -[X²]P
+        // if [X]P = P during the computation, return False
+        if p.is_zero() {
+            true
+        } else {
+            let sigma_p = sigma(p);
+            let mut x_times_p: GroupAffine<_> =
+                p.mul(BigInteger([Parameters0::X[0], 0, 0, 0])).into();
```

Comment:
Let's avoid the inversion here:

```suggestion
            let mut x_times_p: GroupProjective<_> =
                p.mul(BigInteger([Parameters0::X[0], 0, 0, 0]));
```